src/runtime/execution.rs

```diff
@@ -284,10 +276,36 @@ impl ExecutionState {
         F: Future<Output = ()> + Send + 'static,
     {
         Self::with(|state| {
+            let schedule_len = state.current_schedule.len();
+
+            let is_some = state.try_current().is_some();
```

Comment:
Same as the other comment -- `Span`s are just handles, so
```rust
let parent_span = state.try_current().map(|t| t.span.clone()).unwrap_or_else(tracing::Span::current);
```
and then you don't need the branching constructor.